common/v2/features/SendAssets/stateFactory.tsx

```diff
@@ -86,18 +89,22 @@ const TxConfigFactory: TUseApiFactory<State> = ({ state, setState }) => {
 
     provider
       .sendRawTx(signedTx)
-      .then(transactionReceipt => {
+      .then(retrievedTransactionReceipt => {
+        const transactionReceipt = fromTxReceiptObj(retrievedTransactionReceipt);
+        addNewTransactionToAccount(state.txConfig.senderAccount, transactionReceipt || {});
         setState((prevState: State) => ({
           ...prevState,
-          txReceipt: fromTxReceiptObj(transactionReceipt)
+          txReceipt: transactionReceipt
         }));
       })
       .catch(txHash => {
         // If rejected, data is a tx hash, not a receipt. Fetch the receipt, then save receipt for flow
-        provider.getTransactionByHash(txHash).then(transactionReceipt => {
+        provider.getTransactionByHash(txHash).then(retrievedTransactionReceipt => {
+          const transactionReceipt = fromTxReceiptObj(retrievedTransactionReceipt);
+          addNewTransactionToAccount(state.txConfig.senderAccount, transactionReceipt || {});
           setState((prevState: State) => ({
             ...prevState,
-            txReceipt: fromTxReceiptObj(transactionReceipt)
+            txReceipt: transactionReceipt
```

Comment:
FYI: There is also `.finally`
so you can join the save and state update states in a single place:
```
provider.sendRawTx(signedTx)
 .then(retrievedTxReceipt => retrievedTransactionReceipt)
 .catch(txHash => provider.getTransactionByHash(txHash))
 .finally(retrievedTxReceipt => {
    const txReceipt  = fromTxReceiptObj(retrievedTransactionReceipt);
    addNewTransactionToAccount(state.txConfig.senderAccount, txReceipt || {});
    setState((prevState: State) => ({
      ...prevState,
       txReceipt
    }));
 })
```

Reply:
Closing in favour of comment above